lua/bufferline/render.lua

```diff
@@ -113,32 +115,48 @@ local function render(update_names)
     local mod = is_modified and 'Mod' or ''
 
     local separatorPrefix = hl('Buffer' .. status .. 'Sign')
-    local separator = status == 'Inactive' and
-      icons.bufferline_separator_inactive or
-      icons.bufferline_separator_active
+    local separator = is_inactive and
+      icons.separator_inactive or
+      icons.separator_active
 
     local namePrefix = hl('Buffer' .. status .. mod)
-    local name =
-      (not has_icons and state.is_picking_buffer) and
-        slice(buffer_name, 1) or
-        buffer_name
+    local name = buffer_name
 
+    -- Either the buffer name or buffer id
+    local bufferIdPrefix = ''
+    local bufferId = ''
```

Comment:
"buffer id" is ambiguous because I've used "id" in other parts (`state.lua`) to mean "buffer number" (aka the number assigned by vim to each buffer), but here it actually contains the buffer index in the bufferline.